instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_job.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      module Patches
+        # Module to prepend to Resque::Job for instrumentation
+        module ResqueJob
+          def perform
+            job_class = payload_class_name
+
+            attributes = {
+              'messaging.system' => 'resque',
+              'messaging.destination' => queue.to_s,
+              'messaging.destination_kind' => 'queue',
+              'messaging.resque.job_class' => job_class
+            }
+
+            extracted_context = OpenTelemetry.propagation.extract(@payload)
+            OpenTelemetry::Context.with_current(extracted_context) do
+              tracer.in_span("#{queue} process", attributes: attributes, kind: :consumer) do
```

Comment:
Similar to the suggestion in #726, it might be nice to support a config option to name the span after the job class as well.